core/templates/dev/head/services/contextual/UrlService.js

```diff
@@ -31,6 +31,12 @@ oppia.factory('UrlService', ['$window', function($window) {
        in array form. */
     getUrlParams: function() {
       var params = {};
+      /* This is to remove the hash value from URL as mentioned in
```

Comment:
The UrlService should be extracting information from the current URL, and in particular this is a GET method so it shouldn't be actually changing anything.

I think you'll need to:
- Read the current URL
- Compute the current URL without the hash
- Compute the params

without changing what's in the URL bar. Otherwise, this function ends up having side-effects.

Reply:
Ok, so shall I ignore everything after the first occurrence of hash then?

Reply:
Take a look at the Location API: https://developer.mozilla.org/en-US/docs/Web/API/Location

As I've mentioned, you should be getting the data programmatically, not using string manipulation. I recommend writing tests, too.

Reply:
Ok, will do that

Reply:
Done, is this fine? If so, I'll start with the tests.